src/ServiceControl.Transports.RabbitMQ/ConnectionFactory.cs

```diff
@@ -4,13 +4,15 @@
     using System.Net.Security;
     using System.Security.Authentication;
     using System.Security.Cryptography.X509Certificates;
+    using System.Threading;
+    using System.Threading.Tasks;
     using global::RabbitMQ.Client;
 
     class ConnectionFactory
     {
         readonly string endpointName;
         readonly global::RabbitMQ.Client.ConnectionFactory connectionFactory;
-        readonly object lockObject = new object();
+        readonly SemaphoreSlim semaphoreSlim = new(1, 1);
```

Comment:
Don't forget to properly dispose the semaphore

Reply:
Thanks for the heads up.  I'll take another look at this

Reply:
> I thought that the `semaphoreSlim.Release()` did the dispose,

No, it does not. The semaphore is not single use but re-usable.